pkg/helm/release/manager.go

```diff
@@ -286,32 +292,27 @@ func generatePatch(existing, expected runtime.Object) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	ops, err := jsonpatch.CreatePatch(existingJSON, expectedJSON)
+	patchMeta, err := strategicpatch.NewPatchMetaFromStruct(expected)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not generate patch metadata: %w", err)
 	}
-
-	// We ignore the "remove" operations from the full patch because they are
-	// fields added by Kubernetes or by the user after the existing release
-	// resource has been applied. The goal for this patch is to make sure that
-	// the fields managed by the Helm chart are applied.
-	// All "add" operations without a value (null) can be ignored
-	patchOps := make([]jsonpatch.JsonPatchOperation, 0)
-	for _, op := range ops {
-		if op.Operation != "remove" && !(op.Operation == "add" && op.Value == nil) {
-			patchOps = append(patchOps, op)
-		}
+	patch, err := strategicpatch.CreateThreeWayMergePatch(expectedJSON, expectedJSON, existingJSON, patchMeta, true)
```

Comment:
Since both the original and modified documents are `expectedJSON`, how is this different from a two-way merge?

Reply:
A three-way patch ensures that any manual changes made to the resource on cluster (e.g. with `kubectl edit`) that are not explicitly mentioned in the release manifest remain in the resource after this patch is applied.

That is the intent of the current code, so three-way patch was chosen to keep backward-compatibility.